### ISSUE TITLE
Why does the first term in the loss_robust equal F.log_softmax(model(x_adv))?

### ISSUE THREAD

**Issue by changzhang777** [yaodongyu/TRADES#28]
Why does the first term in the loss_robust equal F.log_softmax(model(x_adv))? I have tryed to generate adv samples by noraml PGD, and set the loss to the trades's loss. But I cannot understand why it is and I guess I prefer to use criterion_kl(F.softmax(model(x_adv)), F.softmax(model(x_natural))). Are there some nice people who can answer my question? I would appreciate it.

**Comment by amirhagai** [comment]
As far as I  understand they try to minimize the cross entropy loss, but in torch this loss expect to get one hot vector.
see equation 1 at [https://en.wikipedia.org/wiki/Cross_entropy](url)

**Comment by changzhang777** [comment]
> 

I think you mean that it is uesd to keep consistent with log_softmaxed vector which is in the cross entropy when optimizing. Thx a lot.

**Comment by pingguokiller** [comment]
> Why does the first term in the loss_robust equal F.log_softmax(model(x_adv))? I have tryed to generate adv samples by noraml PGD, and set the loss to the trades's loss. But I cannot understand why it is and I guess I prefer to use criterion_kl(F.softmax(model(x_adv)), F.softmax(model(x_natural))). Are there some nice people who can answer my question? I would appreciate it.

I have the same question. And do you know more about that why the author use "criterion_kl" and "F.log_softmax(model(x_adv))"?
